File: pdlearning/src/microservices/content-microservice/Api/Application/Models/ChapterAttachmentModel.cs
using System;
using ChapterAttachmentEntity = Microservice.Content.Domain.Entities.ChapterAttachment;

namespace Microservice.Content.Application.Models
{
    public class ChapterAttachmentModel
    {
        public ChapterAttachmentModel()
        {
        }

        public ChapterAttachmentModel(ChapterAttachmentEntity chapterAttachment)
        {
            Id = chapterAttachment.Id;
            ObjectId = chapterAttachment.Id;
            FileLocation = chapterAttachment.FileLocation;
            FileName = chapterAttachment.FileName;
            CreatedDate = chapterAttachment.CreatedDate;
            ChangedDate = chapterAttachment.ChangedDate;
            Chapter = new ChapterModel()
            {
                Id = chapterAttachment.Chapter.Id,
                CreatedBy = chapterAttachment.Chapter.CreatedBy,
                CreatedDate = chapterAttachment.Chapter.CreatedDate,
                ChangedBy = chapterAttachment.Chapter.ChangedBy,
                ChangedDate = chapterAttachment.Chapter.ChangedDate,
                ObjectId = chapterAttachment.Chapter.ObjectId,
                OriginalObjectId = chapterAttachment.Chapter.OriginalObjectId,
                Title = chapterAttachment.Chapter.Title,
                Description = chapterAttachment.Chapter.Description,
                TimeStart = chapterAttachment.Chapter.TimeStart,
                TimeEnd = chapterAttachment.Chapter.TimeEnd,
                SourceType = chapterAttachment.Chapter.SourceType
            };
        }

        public Guid Id { get; set; }

        public Guid ObjectId { get; set; }

        public string FileLocation { get; set; }

        public string FileName { get; set; }

        public DateTime CreatedDate { get; set; }

        public DateTime? ChangedDate { get; set; }

        public ChapterModel Chapter { get; set; }
    }
}

File: pdlearning/src/web-apps/src/libraries/domain-api/src/content/models/chapter-attachment.model.ts
export interface IChapterAttachment {
  id: string;
  objectId: string;
  fileLocation: string;
  fileName: string;
  createdDate?: Date;
  changedDate?: Date;
}

export class ChapterAttachment implements IChapterAttachment {
  public id: string;
  public objectId: string;
  public fileLocation: string;
  public fileName: string;
  public createdDate?: Date;
  public changedDate?: Date;
  constructor(data?: IChapterAttachment) {
    if (data == null) {
      return;
    }
    this.id = data.id;
    this.objectId = data.objectId;
    this.fileLocation = data.fileLocation;
    this.fileName = data.fileName;
    this.createdDate = data.createdDate ? new Date(data.createdDate) : null;
    this.changedDate = data.changedDate ? new Date(data.changedDate) : null;
  }
}
